src/main/java/seedu/address/model/assignment/DueDate.java

```diff
@@ -60,8 +75,17 @@ public DueDate(String date, String time) {
      * @param date Date of due date.
      */
     public DueDate(String date) {
-        this(date, LATEST_TIME_IN_DAY);
+        if (isValidFriendlyDate(date)) {
+            this.date = friendlyToDate(date);
+        } else {
+            checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS_DATE);
+            this.date = LocalDate.parse(date, PARSE_DATE_FORMAT);
+        }
+        this.time = LocalTime.parse(LATEST_TIME_IN_DAY, PARSE_TIME_FORMAT);
+        this.dateTime = LocalDateTime.of(this.date, this.time);
+        this.value = this.dateTime.format(OUTPUT_FORMAT);
```

Comment:
I think it is better to keep it as this(date, LATEST_TIME_IN_DAY), as it does the same thing as the 1st constructor, and we can avoid having to edit both constructors if there is something we might have to change in the future